parasoft-findings-teamcity-agent/src/main/java/com/parasoft/findings/teamcity/agent/ParasoftFindingsBuildProcess.java

```diff
@@ -156,63 +160,66 @@ private void doWork() {
             for (File from : reports) {
-                ReportParserDescriptor rpd = getReportParserDescriptor(from);
-                if (rpd != null) {
+                List<ReportParserDescriptor> rpds = getReportParserDescriptors(from);
+                if(rpds.isEmpty()) {
+                    _build.getBuildLogger().warning("Skipping unrecognized report file: " + from.getAbsolutePath());
+                    continue;
+                }
+                for(ReportParserDescriptor rpd : rpds) {
                     _build.getBuildLogger().message("Transforming "+from.getAbsolutePath()+" with "+rpd.getLabel());
-                    String targetFileName = PREFIX + from.getName();
+                    String targetFileName = rpd.getOutputFileNamePrefix() + from.getName();
                     File to = new File(from.getParentFile(), targetFileName);
                     transform(from, to, rpd.getXSL(), checkoutDir);
-                } else {
-                    _build.getBuildLogger().warning("Skipping unrecognized report file: " + from.getAbsolutePath());
                 }
             }
         }
     }
 
-    private ReportParserDescriptor getReportParserDescriptor(File from) {
-        StreamSource xml = new StreamSource(from);
-        XMLEventReader reader = null;
+    private List<ReportParserDescriptor> getReportParserDescriptors(File from) {
+        List<ReportParserDescriptor> descriptors = new ArrayList<ReportParserDescriptor>();
         try {
-            reader = xmlInputFactory.createXMLEventReader(xml);
-            while (reader.hasNext()) {
-                XMLEvent event = reader.nextEvent();
-                switch (event.getEventType()) {
-                case XMLStreamConstants.START_ELEMENT:
-                    String toolName = null;
-                    StartElement startElement = event.asStartElement();
-                    @SuppressWarnings("unchecked")
-                    Iterator<Attribute> attributes = startElement.getAttributes();
-                    while (attributes.hasNext()) {
-                        Attribute attribute = attributes.next();
-                        String name = attribute.getName().getLocalPart();
-                        if ("toolName".equals(name)) {
-                            toolName = attribute.getValue();
-                            break;
-                        }
-                    }
-                    if (toolName != null) {
-                        return "SOAtest".equals(toolName) ?
-                                ReportParserTypes.getDescriptor(ReportParserType.SOATEST.name()) :
-                                ReportParserTypes.getDescriptor(ReportParserType.ANALYZERS.name());
-                    }
-                    return null;
-                default:
-                    break;
-                }
+            Document document = getDocument(from);
+            if(checkIfNodeExists(document, "/ResultsSession/CodingStandards/StdViols/StdViol")) {
```

Comment:
The selection should also consider cases when there's no standard violations but other violations(e.g. Flow/Metric). The logic should be checking whether a child of StdViols which is not DupViol exists.